src/Hazelcast.Net/HazelcastFailoverOptions.cs

```diff
@@ -0,0 +1,90 @@
+
+using System.Collections.Generic;
+using System.Linq;
+using Hazelcast.Configuration.Binding;
+
+namespace Hazelcast
+{
+    /// <summary>
+    /// Represents the Hazelcast client failover options.
+    /// </summary>
+    /// <remarks>
+    /// <para>The client initially tries to connect to the first cluster of the <see cref="Clients"/>
+    /// list, then tries each cluster in the list in order, and cycle the list at most <see cref="TryCount"/>
+    /// times before failing and shutting down.</para>
+    /// <para>When the client gets disconnected, it first tries to reconnect to the current cluster,
+    /// then tries each cluster in the list in order, and cycle the list at most <see cref="TryCount"/>
+    /// times before failing and shutting down.</para>
+    /// <para>So if the <see cref="Clients"/> list is (A, B, C) and client is disconnected from B and
+    /// <see cref="TryCount"/> is 2, it will try C, B, A, C, B, A in this order and then shutdown.</para>
+    /// <para>The retry strategy for each cluster is configured with a 2 minutes timeout, i.e. the
+    /// client will try to connect to each cluster for at most 2 minutes before failing.</para>
+    /// </remarks>
+    public sealed class HazelcastFailoverOptions : HazelcastOptionsBase
+    {
+        /// <summary>
+        /// Gets the Hazelcast failover configuration section name, which is <c>"hazelcast-failover"</c>.
+        /// </summary>
+        internal const string SectionNameConstant = "hazelcast-failover";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HazelcastFailoverOptions"/> class.
+        /// </summary>
+        public HazelcastFailoverOptions()
+        {
+            Clients = new List<HazelcastOptions>();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HazelcastFailoverOptions"/> class.
+        /// </summary>
+        private HazelcastFailoverOptions(HazelcastFailoverOptions other)
+        {
+            TryCount = other.TryCount;
+            Clients = new List<HazelcastOptions>(other.Clients.Select(x => x.Clone()));
+            Enabled = other.Enabled;
+        }
+
+        /// <inheritdoc />
+        internal override string SectionName => SectionNameConstant;
+
+        /// <summary>
+        /// Gets status of failover
+        /// </summary>
+        [BinderIgnore(false)]
+        internal bool Enabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets the number of times that the client will try to reconnect to each
+        /// cluster in the failover setup before shutting down.
+        /// </summary>
+        public int TryCount { get; set; }
+
+        /// <summary>
+        /// Gets the list of cluster in the failover setup.
+        /// </summary>
+        // TODO: consider supporting merging external files?
+        // "clients": [ { "client": "path-to-json" }, ... ]
+        public IList<HazelcastOptions> Clients { get; }
```

Comment:
I have doubts since this is already a client code base. Isn't it a bit weird to name it again? Also, we have removed `Client` prefix for the client states as well.

Reply:
But we have the name `Clients` right now, I don't think `ClientOptions` is that different than the `Clients`